src/ts/greedy-nav/GreedyNav.ts

```diff
@@ -0,0 +1,1062 @@
+    // eslint-disable-next-line no-unused-expressions
+    _this
+        .querySelector<HTMLElement>(navDropdownToggleSelector)
+        ?.setAttribute('priorityNav-count', `${breaks.length}`);
+};
+
+export const updateLabel = (
+    menu: HTMLElement,
+    label: string,
+    navDropdownToggleSelector: string,
+    navDropdownLabelActive: string
+): void => {
+    const toggle = menu.querySelector<HTMLElement>(navDropdownToggleSelector);
+
+    if (toggle == null) {
+        return;
+    }
+
+    toggle.innerHTML = label;
+
+    if (label === navDropdownLabelActive) {
+        toggle.setAttribute('aria-expanded', 'true');
+    } else {
+        toggle.setAttribute('aria-expanded', 'false');
+    }
+};
+
+export const insertAfter = (newNode: Node, referenceNode: Node): void => {
+    if (referenceNode.parentNode !== null) {
+        referenceNode.parentNode.insertBefore(
+            newNode,
+            referenceNode.nextSibling
+        );
+    }
+};
+
+const checkForSymbols = (str: string) => {
+    const firstChar = str.charAt(0);
+    if (firstChar === '.' || firstChar === '#') {
+        return false;
+    }
+    return true;
+};
+
+/**
+ * Get innerwidth without padding
+ * @param element
+ * @returns {number}
+ */
+const getElementContentWidth = (element: HTMLElement) => {
+    const styles = window.getComputedStyle(element);
+    const padding =
+        parseFloat(styles.paddingLeft) + parseFloat(styles.paddingRight);
+
+    return element.clientWidth - padding;
+};
+
+/**
+ * Get viewport size
+ * @returns {{width: number, height: number}}
+ */
+const viewportSize = () => {
+    const doc = document;
+    const w = window;
+    const docEl =
+        doc.compatMode && doc.compatMode === 'CSS1Compat'
+            ? doc.documentElement
+            : doc.body;
+
+    let width = docEl.clientWidth;
+    let height = docEl.clientHeight;
+
+    // mobile zoomed in?
+    if (w.innerWidth && width > w.innerWidth) {
+        width = w.innerWidth;
+        height = w.innerHeight;
+    }
+
+    return { width, height };
+};
+
+/**
+ * Count width of children and return the value
+ * @param e
+ */
+const getChildrenWidth = (e: HTMLElement) => {
+    const children = e.childNodes;
+    let sum = 0;
+    for (let i = 0; i < children.length; i++) {
+        if (children[i].nodeType !== 3) {
+            if (!Number.isNaN((<HTMLElement>children[i]).offsetWidth)) {
+                sum += (<HTMLElement>children[i]).offsetWidth;
+            }
+        }
+    }
+    return sum;
+};
+
+/**
+ * Get width
+ * @param elem
+ * @returns {number}
+ */
+const calculateWidths = (_this: HTMLElement, offsetPixels: number) => {
+    const totalWidth = getElementContentWidth(_this);
+    // Check if parent is the navwrapper before calculating its width
+
+    const restWidth = getChildrenWidth(_this) + offsetPixels;
+    const viewportWidth = viewportSize().width;
+
+    return { totalWidth, restWidth, viewportWidth };
+};
+
+export class GreedyNavMenu {
```

Comment:
Shouldn't we include an interface?

Reply:
I don't think so - it's implicit to the class - what purpose would it serve?